src/query/binary.rs

```diff
@@ -343,12 +345,12 @@ fn scalar_op_scalar(op: BinaryOp, lv: SampleValue, rv: SampleValue) -> SampleVal
         Sub => lv - rv,
 
         // Comparison
-        Eql => (lv == rv) as i64 as SampleValue,
+        Eql => ((lv - rv).abs() < f64::EPSILON) as i64 as SampleValue,
```

Comment:
Oh, stupid me 🙈 Very good catch!

Reply:
Once we'll add clippy to the CI these things won't happen 😌